src/plugins/share/public/components/tabs/export/export_content.tsx

```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useCallback, useEffect, useState } from 'react';
+import { FormattedMessage, injectI18n } from '@kbn/i18n-react';
+
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiCopy,
+  EuiFlexGroup,
+  EuiForm,
+  EuiIcon,
+  EuiModalFooter,
+  EuiRadioGroup,
+  EuiSpacer,
+  EuiSwitch,
+  EuiSwitchEvent,
+  EuiText,
+  EuiToolTip,
+} from '@elastic/eui';
+import useMountedState from 'react-use/lib/useMountedState';
+import { ShareMenuItem } from '../../../types';
+import { type IShareContext } from '../../context';
+
+type ExportProps = Pick<
+  IShareContext,
+  'isDirty' | 'objectId' | 'objectType' | 'onClose' | 'toasts'
+> & {
+  layoutOption?: 'print';
+  aggregateReportTypes: ShareMenuItem[];
+};
+
+type AllowedExports = 'pngV2' | 'printablePdfV2' | 'csv_v2' | 'csv_searchsource' | 'lens_csv';
+
+const ExportContentUi = ({ isDirty, objectType, aggregateReportTypes, toasts }: ExportProps) => {
+  // needed for CSV in Discover
+  const firstRadio =
+    (aggregateReportTypes[0].reportType as AllowedExports) ?? ('printablePdfV2' as const);
+  const [, setIsStale] = useState(false);
+  const [isCreatingReport, setIsCreatingReport] = useState<boolean>(false);
+  const [selectedRadio, setSelectedRadio] = useState<AllowedExports>(firstRadio);
+  const [usePrintLayout, setPrintLayout] = useState(false);
+  const isMounted = useMountedState();
+
+  const markAsStale = useCallback(() => {
+    if (!isMounted) return;
+    setIsStale(true);
+  }, [isMounted]);
+
+  const getProperties = useCallback(() => {
+    if (objectType === 'search') {
+      return aggregateReportTypes[0];
+    } else {
+      return aggregateReportTypes?.filter(({ reportType }) => reportType === selectedRadio)[0];
+    }
+  }, [selectedRadio, aggregateReportTypes, objectType]);
+
+  const handlePrintLayoutChange = useCallback(
+    (evt: EuiSwitchEvent) => {
+      setPrintLayout(evt.target.checked);
+      getProperties();
+    },
+    [setPrintLayout, getProperties]
+  );
+
+  const {
+    generateReportButton,
+    helpText,
+    renderCopyURLButton,
+    generateReport,
+    generateReportForPrinting,
+    downloadCSVLens,
+    absoluteUrl,
+    renderLayoutOptionSwitch,
+  } = getProperties();
+
+  const getRadioOptions = useCallback(() => {
+    if (!aggregateReportTypes.length) {
+      throw new Error('No content registered for this tab');
+    }
+    return aggregateReportTypes.map(({ reportType, label }) => {
+      return { id: reportType, label, 'data-test-subj': `${reportType}-radioOption` };
+    });
```

Comment:
Using some error checking, we can remove the `// ts-ignore` that is below in this file
```suggestion
    return aggregateReportTypes.map(({ reportType, label }) => {
      if (reportType == null) {
        throw new Error('expected reportType to be string!');
      }
      return { id: reportType, label, 'data-test-subj': `${reportType}-radioOption` };
    });
```

Reply:
Oops! I actually saw this error trigger
<img width="941" alt="image" src="https://github.com/elastic/kibana/assets/908371/0965125f-a054-48d0-9639-cdcd3d7549dc">

I wonder, where is `reportType` not a string?

Reply:
Ah, I see: not every share menu item in the share context has a `reportType`
```
shareContext.shareMenuItems.map(i => i.reportType)
> [undefined, 'csv_searchsource']
```

Reply:
Nevermind my last two comments - I messed something up in the registry when testing some cleanups 😅